invenio/modules/knowledge/restful.py

```diff
@@ -386,38 +438,86 @@ def get(self, id):
                          args['match_type'],
                          args['page'], args['per_page'], args['unique'])
 
-    def head(self, id):
+    def head(self, slug):
         """Head knowledge."""
         abort(405)
 
-    def options(self, id):
+    def options(self, slug):
         """Option knowledge."""
         abort(405)
 
-    def post(self, id):
+    def post(self, slug):
         """Post knowledge."""
         abort(405)
 
-    def put(self, id):
+    def put(self, slug):
         """Put knowledge."""
         abort(405)
 
 
+class NotImplementedFooResource(Resource):
+
+    """Path not used in REST API."""
+
+    method_decorators = [
+        error_handler
+    ]
+
+    def get(self, slug, foo, bar=None):
+        """Get."""
+        abort(405)
+
+    def head(self, slug, foo, bar=None):
+        """Head."""
+        abort(405)
+
+    def options(self, slug, foo, bar=None):
+        """Option."""
+        abort(405)
+
+    def post(self, slug, foo, bar=None):
+        """Post."""
+        abort(405)
+
+    def put(self, slug, foo, bar=None):
+        """Put."""
+        abort(405)
+
+
+class NotImplementedFooBarResource(NotImplementedFooResource):
+
+    """Path not used in REST API."""
+
+
 def setup_app(app, api):
     """setup the resources urls."""
     api.add_resource(
+        KnwKBAllResource,
+        '/api/knowledge'
+    )
+    api.add_resource(
         KnwKBResource,
-        '/api/knowledge/<int:id>'
+        '/api/knowledge/<string:slug>'
     )
     api.add_resource(
         KnwKBMappingsResource,
-        '/api/knowledge/<int:id>/mappings'
+        '/api/knowledge/<string:slug>/mappings'
     )
     api.add_resource(
         KnwKBMappingsToResource,
-        '/api/knowledge/<int:id>/mappings/to'
+        '/api/knowledge/<string:slug>/mappings/to'
     )
     api.add_resource(
         KnwKBMappingsFromResource,
-        '/api/knowledge/<int:id>/mappings/from'
+        '/api/knowledge/<string:slug>/mappings/from'
+    )
+
+    # for other urls, return "Method Not Allowed"
+    api.add_resource(
```

Comment:
@jirikuncar @lnielsen 
I used `NotImplementedFooResource` and `NotImplementedFooBarResource` to cover any possible url.
Do you know if exists the possibility to set automatically: if user request any other paths, then returns "Method Not Allowed" (instead of html page)?